source/common/unified_memory_allocation/include/uma/memory_provider.h

```diff
@@ -39,7 +39,8 @@ void umaMemoryProviderDestroy(uma_memory_provider_handle_t hProvider);
 /// \param hProvider handle to the memory provider
 /// \param size number of bytes to allocate
 /// \param alignment alignment of the allocation
-/// \param ptr returns pointer to the allocated memory
+/// \param ptr[in][out] If memory provider is of type UMA_MEMORY_PROVIDER_TYPE_TRACKER, value of *ptr will be used for tracking purposes.
```

Comment:
This is to support provider-less pools (like malloc-based ones).

My idea is to only create provider of type UMA_MEMORY_PROVIDER_TYPE_TRACKER inside umaPoolCreate if the list of providers is empty (and forbid user from passing tracker with this type) so we can be sure that UMA_MEMORY_PROVIDER_TYPE_TRACKER  provider will handle this properly but still, this feels like a hack. Maybe you have some better ideas?

Reply:
We could just expose memoryTrackerAdd/Remove functions - it's more explicit but it might be harder to 'enforce' tracking the memory. With `UMA_MEMORY_PROVIDER_TYPE_TRACKER ` the pool at least gets an instance that is supposed to be used for tracking.

Reply:
From the user perspective, I think UMA itself should be responsible for memory tracking (not the user). And we should try to hide tracking functionality inside UMA and do not expose anything related to memory tracking.

Reply:
Right, this is just a special case for when pool does not use any memory provider for allocations. I agree it would be much easier if we just require having at least one memory provider for data.